[Merged by Bors] - chore: an injective bounded linear operator has closed range iff it is anti-Lipschitz

grunweg opened the pull request:
And make the argument `f` of `ContinuousLinearMap.equivRange` implicit,
as the explicit hypotheses `hf` and `hclo` mention `f`.

---

This is motivated by defining immersions between Banach manifolds: the correct condition is that the `mfderiv` is a split linear map, and proving that the composition of split linear maps is split requires this lemma.

~There is one remaining sorry, which I don't know how to do: @winstonyin as the original author of `ContinuousLinearMap.equivRange`, do you happen to have an idea?~ `RubenVandeVelde` helped me fix the sorry, but better approaches are very welcome.

<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

grunweg commented:
I guess [ContinuousLinearMap.closed_complemented_range_of_isCompl_of_ker_eq_bot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Normed/Operator/Banach.html#ContinuousLinearMap.closed_complemented_range_of_isCompl_of_ker_eq_bot) mostly solves (and is just badly named). Hm.

jcommelin commented on Mathlib/Analysis/Normed/Operator/Banach.lean:
```diff
@@ -364,6 +364,54 @@ theorem coe_equivRange (f : E →SL[σ] F) (hinj : Injective f) (hclo : IsClosed
+
+/-- An injective bounded linear operator between Banach spaces has closed range
+iff it is anti-Lipschitz. -/
+lemma isClosed_range_if_antilipschitz_of_injective (f : E →L[𝕜] F)
```
```suggestion
lemma isClosed_range_iff_antilipschitz_of_injective (f : E →L[𝕜] F)
```

jcommelin commented on Mathlib/Analysis/Normed/Operator/Banach.lean:
```diff
@@ -364,6 +364,54 @@ theorem coe_equivRange (f : E →SL[σ] F) (hinj : Injective f) (hclo : IsClosed
   rfl
 
+@[simp]
+lemma equivRange_symm (f : E →SL[σ] F) (hinj : Injective f) (hclo : IsClosed (range f)) :
```
```suggestion
lemma equivRange_symm_toLinearEquiv (f : E →SL[σ] F) (hinj : Injective f) (hclo : IsClosed (range f)) :
```

grunweg commented:
Addressing the review comment made me realise that presumable `f` should be implicit, since the hypotheses `hf` and `hclo` mention it. I just pushed a commit making it so. @jcommelin Let me know if you disagree or are not sure, then I can roll that back/start a separate discussion.

grunweg commented:
Let me just merge this; if there is any controversy about the explicitness change, I'm happy to revert or tweak this.
bors merge
